### PR TITLE
Allow exported Maven artifacts to have a template pom

### PR DESCRIPTION
**Pull request by shs96c**
Without this template, there is not enough information available
within the build files to correctly generate a valid and useful,
and more importantly, valid `pom.xml`.

Selenium uses this functionality extensively within its own build
in order to successfully export artifacts to Maven Central.

### CODE REVIEW

**Comment by facebook-github-bot-9** [comment]
By analyzing the blame information on this pull request, we identified @sdwilsh, @aiked and @yiding to be potential reviewers.

**Comment by shs96c** [comment]
For an example of this in action:

https://github.com/SeleniumHQ/selenium/blob/selenium-3.0.0-beta-3/java/client/src/org/openqa/selenium/BUCK#L23

File: src/com/facebook/buck/maven/Pom.java [hunk 1]
```diff
@@ -114,27 +137,212 @@ public Artifact apply(HasMavenCoordinates input) {
...
-      try {
-        ModelBuildingResult modelBuildingResult = modelBuilder.build(modelBuildingRequest);
+      Model fromFile = constructModel(file, model);
+      model = merge(model, fromFile);
+    }
+
+    return model;
+  }
+
+  private Model merge(Model first, @Nullable Model second) {
```
Comment (Coneko) [inline src/com/facebook/buck/maven/Pom.java]: This method is massive, is there really no method in the library to do this?

Comment (shs96c) [inline src/com/facebook/buck/maven/Pom.java]: I can't find a public method that does this. Most irksome. It's not regular enough to use the JavaBeans APIs to introspect over it either.

**Comment by sdwilsh** [comment]
Can you add a test to handle the case where you pass this in as well?  It looks like all tests you touched just handle the `Optional.absent()` case.

**Comment by facebook-github-bot-2** [comment]
@shs96c updated the pull request - [view changes](https://github.com/facebook/buck/pull/869/files/db2bd46f555b2386a8072862ae0f25c11073bf77..81e3e04813e593ab7ccc2008f7280c553cf2d122)

**Comment by facebook-github-bot-7** [comment]
@shs96c updated the pull request - [view changes](https://github.com/facebook/buck/pull/869/files/81e3e04813e593ab7ccc2008f7280c553cf2d122..87b6a6902b993131c2358711182a527f8b3e1e04)

**Comment by shs96c** [comment]
Added a test

**Pull request closed without merging**